app/main.py

```diff
@@ -1,2 +1,56 @@
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_size = 8
+        self.hash_table = [[] for _ in range(self.hash_size)]
+        self.treshold = int(self.hash_size * 2 / 3)
+
+    def resize_hash(self) -> None:
+        old_hash_table = self.hash_table
+        self.hash_size *= 2
+        self.treshold = int(self.hash_size * 2 / 3)
+        self.length = 0
+        self.hash_table = [[] for _ in range(self.hash_size)]
+        for item in old_hash_table:
+            if item:
+                self.__setitem__(item[0], item[1])
+
+    def __setitem__(self, key: object, value: object) -> None:
+        if self.length == self.treshold:
+            self.resize_hash()
+        index = hash(key) % self.hash_size
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, value]
+                self.length += 1
+                return
+            if key in self.hash_table[index]:
```

Comment:
you should also check the hash

Reply:
But I did it at the beginning of this method, should I do the same inside the loop, or just move it from the begging to the loop? Cause, then it will be a duplicate of the code

Reply:
you should check the hash, now you check only the key, add another condition to this `if`